src/bid.rs

```diff
@@ -112,11 +73,15 @@ impl Bid {
         ]);
     }
 
-    pub fn prove_score_generation(&self, composer: &mut StandardComposer) -> Result<Proof, Error> {
+    pub fn prove_score_generation(
+        &self,
+        composer: &mut StandardComposer,
+    ) -> Result<Proof, Error> {
         use crate::score_gen::score::prove_correct_score_gadget;
 
         prove_correct_score_gadget(composer, self)?;
-        // XXX: Return the proof with a pre-computed PreprocessedCircuit and ProverKey
+        // TODO: Return the proof with a pre-computed PreprocessedCircuit and
+        // ProverKey
         unimplemented!()
```

Comment:
It is unclear to me the not met blocker that justify this `unimplemented`

Reply:
Not sure about it, I just port in this case the code of Carlos and was going to check in a follow up review.
I can dig this a bit more to figure out what was the original intent.

Reply:
I think this may have been left unimplemented because Carlos was waiting for the Prover/Verifier abstraction, along with the other PLONK API refactors .

Reply:
Do be on the safe side, since Carlos is coming back next week, my suggestion is to land this code as is; and open a follow up issue – and assign it to Carlos.
I'm afraid otherwise we might spend more time than it's needed to figure what Carlos had in mind.

Reply:
As agreed I filed #23 and assigned to @CPerezz , so we can merge this one.

Reply:
@kevaundray was right.